bdk_core_example/src/main.rs

```diff
@@ -245,14 +245,19 @@ fn main() -> anyhow::Result<()> {
         Commands::Sync { unused } => {
             let mut spks = vec![];
             if unused {
-                spks.extend(tracker.iter_unused());
-            }
-            let spk_iter = spks.into_iter().map(|(index, script)| {
-                eprintln!("scanning {:?}", index);
-                script.clone()
-            });
+                spks.extend(tracker.iter_unused().into_iter().map(|(index, script)| {
+                    eprintln!("scanning {:?}", index);
+                    script.clone()
+                }));
+            } else {
+                spks.extend(tracker.script_pubkeys().into_iter().map(|(index, script)| {
+                    eprintln!("scanning {:?}", index);
+                    script.clone()
+                }));
```

Comment:
There should be two scanning options: `unused` and `unspent`. `unspent` should add the spks of all unspent utxos to the list. If nothing is specified just assume `unused=true`, and `unspent=true`. We can add an `--all` option to scan all spks given out too (what you implemented) if we want as well but that should definitely be opt in.

Reply:
Okay got it

Reply:
What you just pushed looks good but `--all` necessarily includes all the `spks` from unspent and unsued so there's no need to add those explicitly if `--all` is set.

Reply:
By default now, it will scan all and only scan `unused` or `unspent` when explicitly requested